ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt

```diff
@@ -120,4 +121,79 @@ interface UtxoLedgerService {
         session: FlowSession,
         validator: UtxoTransactionValidator
     ): UtxoSignedTransaction
+
+    /**
+     * Sends a transaction builder to another session with all dependent back chains.
+     *
+     * @param transactionBuilder The [UtxoTransactionBuilder] to send.
+     * @param session The receiver [FlowSession].
+     */
+    @Suspendable
+    fun sendTransactionBuilder(
+        transactionBuilder: UtxoTransactionBuilder,
+        session: FlowSession
+    )
+
+    /**
+     * Receives a transaction builder from another session.
+     *
+     * @param session The [FlowSession] to receive the [UtxoTransactionBuilder] from.
+     */
+    @Suspendable
+    fun receiveTransactionBuilder(
+        session: FlowSession,
+    ): UtxoTransactionBuilder
+
+    /**
+     * Sends the differences of transaction builders to another session with all dependent back chains.
+     *
+     * @param transactionBuilder The [UtxoTransactionBuilder] to send.
+     * @param session The receiver [FlowSession].
+     * @param originalTransactionalBuilder Baseline [UtxoTransactionBuilder] to calculate the differences to be sent.
+     */
+    @Suspendable
+    fun sendTransactionBuilderDiff(
+        transactionBuilder: UtxoTransactionBuilder,
+        session: FlowSession,
+        originalTransactionalBuilder: UtxoTransactionBuilder
+    )
+
+    /**
+     * Receives a transaction builder from another session then appends its parts to an existing one.
+     *
+     * @param session The [FlowSession] to receive the [UtxoTransactionBuilder] from.
+     * @param originalTransactionalBuilder The received parts will be appended to this [UtxoTransactionBuilder].
+     */
+    @Suspendable
+    fun receiveAndUpdateTransactionBuilder(
```

Comment:
I don't like that name - I think it should be `receiveTransactionBuilderDiff()`, to be symmetrical with the sending function. It should still apply the diff to a passed in tx builder.

Reply:
I have renamed it.